File: this-is-coding-test/8.다이나믹_프로그래밍/3.개미_전사.py
input_data = [1, 3, 1, 5]
input_data_length = len(input_data)

d = [0] * 100

d[0] = input_data[0]
d[1] = max(input_data[0], input_data[1])

for i in range(2, input_data_length):
    d[i] = max(d[i-2] + input_data[i], d[i-1])

print(max(d[input_data_length-1], d[input_data_length]))

File: this-is-coding-test/8.다이나믹_프로그래밍/3.개미_전사.js
const input_data = [1, 5, 1, 5, 1, 1, 5, 1, 1, 5, 1, 5, 1, 5];
const cache = [input_data[0], Math.max(input_data[0], input_data[1])];

function func(x) {
  cache[x] = Math.max(cache[x - 1], cache[x - 2] + input_data[x]);
}

for (let i = 2; i < input_data.length; i++) {
  func(i);
}

console.log(cache);
